### PR TITLE
Feat/newsletter

### LINKED ISSUE #634: ✨ Enhancement: Newsletter landing page
### Is your feature request related to a problem? Please describe

As part of #308 we implementer an initial version of the newsletter component that is now available in the branch [web-release-3](https://github.com/json-schema-org/website/tree/web-release-3) pending to be merged to main alongside other features part of the[ release 3 of the docs strategy](https://github.com/json-schema-org/website/issues/158).

That initial version of the page available in /pages/newsletter is only providing a subscribe form but we'd like to make that page much more powerful by showing also a welcome message and the previously published newsletter. I do like what the CNCF did with their newsletter: https://www.cncf.io/kubeweekly/

I'd like to have something similar. The newsletter will be created in mailchimp so probably we'll need to have a json or yaml file inside the /data folder with the details of newsletters:

- url
- title
- Date

I do like the design of CNCF so let's do something similar. In this case I think we can use the generic card component we build using Title, Body and Link.

### Describe the solution you'd like

See the feature description

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### Are you working on this?

No

### PR DESCRIPTION
**Pull request by Akshaybagai52**
**What kind of change does this PR introduce?**

**Issue Number:**

-  Closes #634 
-  Related to #___ 
-  Others? 

**Screenshots/videos:**

**If relevant, did you update the documentation?**

**Summary**

**Does this PR introduce a breaking change?**

### CODE REVIEW

**Review by benjagm — CHANGES_REQUESTED** [review]
Great first step!!

Now we have the data and the functionality and we only need to improve design and user experience:
- Improve the hero section to replace the current subscribe form for a two columns design like [Kubeweekly](https://www.sktime.net/en/latest/get_involved/mentoring.html).
Column 1 : title, subtitle and link to submit content | Column 2 : Subscribe form.
Screenshot 2024-04-30 at 15 03 41
- The area to show the newsletters require some changes:
  - Remove the title "Previous JSON Schema Newsletter" and replace it for grouping by Year, in this case "2024". The left margin/padding needs to be fixed.
  - Better to change the background color of this area with a light grey.
  - The newsletters needs to be sorted in descending order from newest to oldest.
  
 Great progress here!! Thanks!!

File: components/NewsletterCard.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+
+type NewsletterProps = {
+  url: string;
+  title: string;
+  date: string;
```
Comment (DhairyaMajmudar) [inline components/NewsletterCard.tsx:6]: Hello @Akshaybagai52,  
Here how's about using `Date` data type instead of `string`.

Comment (Akshaybagai52) [reply components/NewsletterCard.tsx:6]: Yes I will fix this

**Comment by DhairyaMajmudar** [comment]
Thank you @Akshaybagai52 can you pls. change the destiny branch to `web-release-newsletter` : )

**Comment by Akshaybagai52** [comment]
> Thank you @Akshaybagai52 can you pls. change the destiny branch to `web-release-newsletter` : )

Okay

**Comment by Akshaybagai52** [comment]
> Thank you @Akshaybagai52 can you pls. change the destiny branch to `web-release-newsletter` : )

I've changed the destiny branch

**Comment by benjagm** [comment]
Hi Akshay there is a lot of unnecessary files in the PR. I think the best way to proceed to avoid this will be:

1. Create a fork from the branch web-release-newsletter
2. Make sure you manually add the new files of the newsletter feature.
3. Create the PR

Otherwise it will be difficult to integrate. I know that this was not supposed to be needed but finally we needed to change the release date of the newsletter and remove from release 3.

Thanks a lot!

**Comment by benjagm** [comment]
Closed in favour of #733

**Pull request closed without merging**